addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -59,7 +59,7 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
      */
     _handleCarrierUpdateResult: function (result) {
         this._handleCarrierUpdateResultBadge(result);
-        var $payButton = $('#o_payment_form_pay .monetary_field');
+        var $payButton = $('#o_payment_form_pay');
```

Comment:
You should use the same selectore as in website_sale_payment: $('button#o_payment_form_pay')